### PR TITLE
Created Shell_Sort.php

### LINKED ISSUE #2156: Shell Sort
- [x] C
- [x] C++
- [x] C# - @Harikrishnan6336
- [x] CoffeeScript
- [ ] Dart
- [x] Go - @lakshyajit165
- [x] Java
- [x] JS
- [ ] Kotlin - @myselfanuj
- [x] Php - @saloniankita
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by saloniankita**
Fixes #2156 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Shell Sort in PHP

#### Languages Used:
- PHP

#### Files Added:
- Shell_Sort.php

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Shell_Sort/Shell_Sort.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+function shell_Sort($new_arr)
+{   
+    $a = round(count($new_arr)/2);    //Gap Value
```
Comment (singh-shreya6) [inline Shell_Sort/Shell_Sort.php:4]: Please add space around operators. Here /.

File: Shell_Sort/Shell_Sort.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    while($a > 0)
+    {
+        for($b = $a; $b < count($new_arr);$b++)    //Comparing and Sorting the elements of array
```
Comment (singh-shreya6) [inline Shell_Sort/Shell_Sort.php:7]: Please add after second semi colon in this line

File: Shell_Sort/Shell_Sort.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            $d = $new_arr[$b];
+            $c = $b;
+            while($c >= $a && $new_arr[$c-$a] > $d)
```
Comment (singh-shreya6) [inline Shell_Sort/Shell_Sort.php:11]: Please add space around -

File: Shell_Sort/Shell_Sort.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                $new_arr[$c] = $new_arr[$c - $a];
+                $c -= $a;
+            }
+            $new_arr[$c] = $d;
+        }   
+        $a = round($a/2.2);
```
Comment (singh-shreya6) [inline Shell_Sort/Shell_Sort.php:18]: Please add space around /

File: Shell_Sort/Shell_Sort.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+   }   
+    return $new_arr;    //Array is returned after sorting
+}   
+ 
+//Driver Code :
+$sample_arr = array(4, 9, 11, -5, 8, 1, 0, 6, -3);
+echo "Array before sorting :<br/>";
+echo implode(', ',$sample_arr);
```
Comment (singh-shreya6) [inline Shell_Sort/Shell_Sort.php:26]: Please add space after comma

File: Shell_Sort/Shell_Sort.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+echo "<br/>";
+echo "<br/>Array after sorting :<br/>";
+echo implode(', ',shell_Sort($sample_arr)). PHP_EOL;
```
Comment (singh-shreya6) [inline Shell_Sort/Shell_Sort.php:29]: Please add space after comma

**Comment by saloniankita** [comment]
I've made the changes requested. Kindly check @singh-shreya6.

**Comment by saloniankita** [comment]
I'm facing issues in squashing the commits. Can you please help me in this ?

**Comment by singh-shreya6** [comment]
As shared by one of the participants, please follow this:
Suppose the name of my working branch is my_branch

I will switch to my branch
git checkout my_branch

After that, I will do all my work in that branch. I will modify files, work on them and finally push all the commits and changes.

Now I need to squash the commits [right now I am in my_branch]
git rebase -i upstream/master

A screen will open up showing the list of all commits with different labels along with it.
Keep the first commit as pick
& change all other commits to squash

This will squash (merge) the respective commit onto previous one
After changing all labels to squash except the first one, save & exit.

Next, a screen will appear showing all commit messages, If you want, you can change the commit messages or leave them as it is. Save and exit.

This will rebase all the commits in case there are no merge conflicts.

Last step is to force push the commits so that commit count becomes one.
git push --force origin my_branch

Done.

**Comment by singh-shreya6** [comment]
@saloniankita Squash?

**Comment by saloniankita** [comment]
@singh-shreya6 Done

**Pull request merged**